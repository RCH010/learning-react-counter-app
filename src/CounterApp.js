import React, {useState} from 'react';
import PropTypes from 'prop-types'

const CounterApp = ({value = 10}) => {

    const [counter, setCounter] = useState(value);
    
    //handleAdd
    const handleAdd = () => setCounter((c) => c + 1);
    const handleminus = () => setCounter((c) => c - 1);
    const handleReset = () => setCounter((c) => value);
  
    return (<>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        <button onClick={handleAdd}>add 1</button>
        <button onClick={handleReset}>Reset</button>
        <button onClick={handleminus}>minus 1</button>
    </>);
};


CounterApp.propType = {
    value: PropTypes.number
}
export default CounterApp;

