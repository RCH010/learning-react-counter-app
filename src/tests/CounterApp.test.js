import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('CounterApp tests', () => {
    let wrapper = shallow(<CounterApp/>);
    // esta funciÃ³n se ejecuta antes de cada una de las pruebas
    beforeEach(() => {
        wrapper = shallow(<CounterApp/>);
    })

    test('should display CounterApp correctly w/default values', () => {
        expect(wrapper).toMatchSnapshot();
        const defaultValue = wrapper.find('h2').text();
        expect(defaultValue).toBe("10");
    })
    
    test('should CounterApp display correcte value', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>);
        const defaultValue = wrapper.find('h2').text();
        expect(defaultValue).toBe(value.toString());
    })

    test('should increment 1, in counterApp', () => {
        const btn1 = wrapper.find('button').at(0);
        btn1.simulate('click');

        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('11');
    })

    test('should decrement 1, in counterApp', () => {
        const btn1 = wrapper.find('button').at(2);
        btn1.simulate('click');

        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('9');
    })
    
    test('should reset default value when reset', () => {
        const value = 110;
        const wrapper = shallow(<CounterApp value={value}/>);
        // simulate -1
        wrapper.find('button').at(2).simulate('click');
        // reset value
        wrapper.find('button').at(1).simulate('click');

        expect(wrapper.find('h2').text()).toBe(value.toString());
    })
    
})
