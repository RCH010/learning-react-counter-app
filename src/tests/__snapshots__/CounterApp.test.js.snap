// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp tests should display CounterApp correctly w/default values 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    add 1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    minus 1
  </button>
</Fragment>
`;
